autoload/lsp/ui/vim/completion.vim

```diff
@@ -0,0 +1,254 @@
+        \ || !l:capabilities['completionProvider']['resolveProvider']
+    return a:completion_item
+  endif
+
+  let l:ctx = {}
+  let l:ctx['response'] = {}
+  function! l:ctx['callback'](data) abort
+    let l:self['response'] = a:data.response
+  endfunction
+
+  call lsp#send_request(a:server_name, {
+        \   'method': 'completionItem/resolve',
+        \   'params': a:completion_item,
+        \   'sync': 1,
+        \   'on_notification': function(l:ctx['callback'], [], l:ctx)
+        \ })
+
+  if empty(l:ctx['response'])
+    return a:completion_item
+  endif
+
+  if lsp#client#is_error(l:ctx['response'])
+    return a:completion_item
+  endif
+
+  if empty(l:ctx['response']['result'])
+    return a:completion_item
+  endif
+
+  return l:ctx['response']['result']
+endfunction
+
+"
+" Remove inserted text during completion.
+"
+function! s:clear_inserted_text(line, position, completed_item, completion_item) abort
+  " Remove commit characters.
+  call setline('.', a:line)
+
+  " Create range to remove v:completed_item.
+  let l:range = {
+        \   'start': {
+        \     'line': a:position[1] - 1,
+        \     'character': lsp#utils#to_char('%', a:position[1], a:position[2] + a:position[3]) - strchars(a:completed_item['word'])
+        \   },
+        \   'end': {
+        \     'line': a:position[1] - 1,
+        \     'character': lsp#utils#to_char('%', a:position[1], a:position[2] + a:position[3])
+        \   }
+        \ }
+
+  " Expand remove range to textEdit.
+  if has_key(a:completion_item, 'textEdit')
+    let l:range['start']['character'] = min([
+          \   l:range['start']['character'],
+          \   a:completion_item['textEdit']['range']['start']['character']
+          \ ])
+    let l:range['end']['character'] = max([
+          \   l:range['end']['character'],
+          \   a:completion_item['textEdit']['range']['end']['character']
+          \ ])
+  endif
+
+  " Remove v:completed_item.word (and textEdit range if need).
+  call lsp#utils#text_edit#apply_text_edits(lsp#utils#get_buffer_uri(bufnr('%')), [{
+        \   'range': l:range,
+        \   'newText': ''
+        \ }])
+
+  " Move to complete start position.
+  call cursor(
+        \   l:range['start']['line'] + 1,
+        \   lsp#utils#to_col('%', l:range['start']['line'] + 1, l:range['start']['character'])
+        \ )
+endfunction
+
+"
+" Get textEdit.newText or insertText when the text is not same to v:completed_item.word.
+"
+function! s:get_expand_text(completed_item, completion_item) abort
+  let l:text = a:completed_item['word']
+  if has_key(a:completion_item, 'textEdit')
+    let l:text = a:completion_item['textEdit']['newText']
+  elseif has_key(a:completion_item, 'insertText')
+    let l:text = a:completion_item['insertText']
+  endif
+  return l:text != a:completed_item['word'] ? l:text : ''
+endfunction
+
+"
+" Expand text
+"
+function! s:expand_text_simply(text) abort
+  let l:pos = {
```

Comment:
we should have `let l:pos = lsp#utils#position#_vim_to_lsp('%', line('.'), col('.'))`